homeassistant/components/tractive/binary_sensor.py

```diff
@@ -29,35 +29,36 @@ class TractiveBinarySensor(TractiveEntity, BinarySensorEntity):
     """Tractive sensor."""
 
     def __init__(
-        self, user_id: str, item: Trackables, description: BinarySensorEntityDescription
+        self,
+        client: TractiveClient,
+        item: Trackables,
+        description: BinarySensorEntityDescription,
     ) -> None:
         """Initialize sensor entity."""
-        super().__init__(user_id, item.trackable, item.tracker_details)
+        super().__init__(client.user_id, item.trackable, item.tracker_details)
 
         self._attr_unique_id = f"{item.trackable['_id']}_{description.key}"
-        self.entity_description = description
-
-    @callback
-    def handle_server_unavailable(self) -> None:
-        """Handle server unavailable."""
         self._attr_available = False
-        self.async_write_ha_state()
+        self.entity_description = description
+        self._client = client
 
     @callback
-    def handle_hardware_status_update(self, event: dict[str, Any]) -> None:
-        """Handle hardware status update."""
+    def handle_status_update(self, event: dict[str, Any]) -> None:
+        """Handle status update."""
         self._attr_is_on = event[self.entity_description.key]
-        self._attr_available = True
-        self.async_write_ha_state()
+
+        super().handle_status_update(event)
 
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
+        if not self._client.subscribed:
+            self._client.subscribe()
```

Comment:
Instead of adding this two lines on each entity, it is possibile to move this two lines into `TrackerEntity` and than call here `super().async_added_to_hass()`
What do you think?

Reply:
Yeah good point, I missed that one. I’ll update the PR tomorrow.

Reply:
It's crazy but with this change the events for different sensor classes get mixed up and I get a `KeyError` when setting native_value. What am I missing?
```
2023-08-05 15:08:19.750 ERROR (MainThread) [homeassistant.util.logging] Exception in handle_status_update when dispatching 'tractive_tracker_activity_updated-60c1e02264eac4ca8f7de273': ({'minutes_active': 96, 'daily_goal': 120},)
Traceback (most recent call last):
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/tractive/sensor.py", line 79, in handle_status_update
    self._attr_native_value = event[self.entity_description.key]
                              ~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'calories'
```

Reply:
I must have made some mistake. I started from scratch, simplified the whole sensor platform and now everything works fine.

Reply:
I'm sorry for not getting back to you sooner. Perfect that you have fixed it in the meantime :)